file_catalog/server.py

```diff
@@ -99,225 +94,207 @@ def set_last_modification_date(metadata: types.Metadata) -> None:
+            logger.info(f"config: {key} => '{config[key]}'")
         else:
-            cookie_secret = ''.join(chr(random.randint(0, 128)) for _ in range(16))
-
-        app = tornado.web.Application(
-            [
-                (r"/", MainHandler, main_args),
-                (r"/login", LoginHandler, main_args),
-                (r"/account", AccountHandler, main_args),
-                (r"/api", HATEOASHandler, api_args),
-                (r"/api/files", FilesHandler, api_args),
-                (r"/api/files/count", FilesCountHandler, api_args),
-                (r"/api/files/([^\/]+)", SingleFileHandler, api_args),
-                (r"/api/files/([^\/]+)/actions/remove_location", SingleFileActionsRemoveLocationHandler, api_args),
-                (r"/api/files/([^\/]+)/locations", SingleFileLocationsHandler, api_args),
-                (r"/api/collections", CollectionsHandler, api_args),
-                (r"/api/collections/([^\/]+)", SingleCollectionHandler, api_args),
-                (r"/api/collections/([^\/]+)/files", SingleCollectionFilesHandler, api_args),
-                (r"/api/collections/([^\/]+)/snapshots", SingleCollectionSnapshotsHandler, api_args),
-                (r"/api/snapshots/([^\/]+)", SingleSnapshotHandler, api_args),
-                (r"/api/snapshots/([^\/]+)/files", SingleSnapshotFilesHandler, api_args),
-            ],
-            static_path=static_path,
-            template_path=template_path,
-            log_function=tornado_logger,
-            login_url='/login',
-            xsrf_cookies=True,
-            cookie_secret=cookie_secret,
-            debug=debug,
-        )
-        app.listen(port)
-
-    def run(self) -> None:  # pylint: disable=R0201
-        """Start IO loop."""
-        tornado.ioloop.IOLoop.current().start()
-
-
-# --------------------------------------------------------------------------------------
-# Main Routes (unused)
-# --------------------------------------------------------------------------------------
+            logger.info(f"config: {key} => 'REDACTED'")
+    logger.info(f"port: {port}")
+    logger.info(f"debug: {debug}")
+
+    static_path = get_pkgdata_filename('file_catalog', 'data/www')
+    if static_path is None:
+        raise Exception('bad static path')
+
+    template_path = get_pkgdata_filename('file_catalog', 'data/www_templates')
+    if template_path is None:
+        raise Exception('bad template path')
+
+    args_dict = {
+        "auth": {
+            "audience": config["AUTH_AUDIENCE"],
+            "openid_url": config["AUTH_OPENID_URL"],
+        },
+        "base_url": "/api",
+        "config": config,
+        "db": mongo,
+        "debug": debug,
+    }
+    if bool(os.environ.get('CI_TEST_ENV', False)):
+        del args_dict["auth"]
+    args = RestHandlerSetup(args_dict)  # type: ignore
+    args["config"] = config
+    args["db"] = mongo
+
+    cookie_secret = ''.join(random.choice('0123456789abcdef') for _ in range(64))
```

Comment:
I see what you're doing. It's clever certainly, but something like `hex(random_int)` maybe be clearer.

Reply:
Actually, the new `secrets` module makes this much easier:
```python
cookie_secret = secrets.token_hex(16)
```

Reply:
Excellent. I like this new `secrets` module.